modules/core/src/main/scala/org/tessellation/domain/statechannel/StateChannelService.scala

```diff
@@ -0,0 +1,31 @@
+package org.tessellation.domain.statechannel
+
+import cats.data.NonEmptyList
+import cats.data.Validated.{Invalid, Valid}
+import cats.effect.kernel.Async
+import cats.syntax.all._
+
+import org.tessellation.domain.aci.StateChannelOutput
+import org.tessellation.domain.cell.{L0Cell, L0CellInput}
+import org.tessellation.domain.statechannel.StateChannelValidator.StateChannelValidationError
+
+trait StateChannelService[F[_]] {
+  def process(stateChannel: StateChannelOutput): F[Either[NonEmptyList[StateChannelValidationError], Unit]]
```

Comment:
Looks like a good candidate for `ValidatedNel` because it is technically `Validated[NonEmptyList[E], R]`.

Especially that `stateChannelValidator` returns `Validated` so you can just use `Applicative` to aggregate errors into `ValidateNel`

Reply:
I thought the same @kpudlik, but from the other side, it exposes that processing either succeeded or failed (and reason why), so imo it's fine as it is 🤔